kanidmd/src/lib/event.rs

```diff
@@ -558,16 +558,11 @@ impl ModifyEvent {
     pub fn from_internal_parts(
         _audit: &mut AuditScope,
         ident: Identity,
-        target_uuid: Uuid,
         ml: &ModifyList<ModifyInvalid>,
-        filter: Filter<FilterInvalid>,
+        filter: &Filter<FilterInvalid>,
         qs: &QueryServerWriteTransaction,
     ) -> Result<Self, OperationError> {
-        let f_uuid = filter_all!(f_eq("uuid", PartialValue::new_uuid(target_uuid)));
```

Comment:
It's related to `ModifyEvent::from_internal_parts` in `kanidmd/src/lib/actors/v1_write.rs` which previously was prescriptive (you need to supply a single target_uuid that we want to modify, and we'll turn that into a filter) but now it allows providing of a filter to assemble tho modification instead (so that we can use it in handle_internalpatch again an arbitrary filter derived from the https front end).

Reply:
> > I'm still a little confused where Kanidm comes into the Oauth2 process. We're both the client and the server? Are we the Auth server but not the resource server? What do we do differently from the [`oauth2 crate`](https://docs.rs/oauth2/4.0.0/oauth2/index.html)?
> 
> Right, oauth2 has three parts:
> 
>     * Client, the browser where a user is sitting
> 
>     * Resource Server, the system or resource you want to access
> 
>     * Identity Provider, the holder of who you are and decided if you have access.
> 
> 
> Oauth2 is a system that allows a client, when accessing a resource server, to be redirected to the IDP to make authorisation decisions on behalf of the resource server.
> 
> In this way, Kanidm will be the IDP.

IDP = Identity provider, correct?
> 
> We also need to implement some client and resource server behaviour for _testing_ the behaviour of our IDP.

So stuff in `kanidm_client` isn't really meant to be deployed?
> 
> The oauth2 crate is extremely "opinionated" and only implements the components to be a resource server. It is not possible to use it to be a IDP or Client. Additionally, even for our use as an RS, because it's so opinionated it makes it more complex to try and use it as it implements so much magic behind the scenes, that it's a lot simpler to just bypass that crate all together.
> 
> edit: you may want to look at https://github.com/kanidm/kanidm/blob/master/designs/oauth.rst for an extended discussion

I'll have a look!
> 
> Does that help?

Yes, thank you!

Reply:
IDP is identity provider, yep.

The kanidm_client parts are api calls for administering the resource servers that are connected to our IDP. It's NOT about doing authorisation calls/checks. There are also some parts in kanidm_client/tests/ that check that behaviour, but are again, not deployed.